### ISSUE TITLE
Workflow when creating new pages, then viewing them

### ISSUE THREAD

**Issue by paulhibbitts** [getgrav/grav-plugin-admin#232]
While going through some basic page creation workflows with the latest RC candidate, I am not getting pages to be displayed as I would expect during the creation process - here is the process I am trying:

1) Create a new page, either root or a child of existing page
2) Enter content of page, etc.
3) Check "published" for each page

At this point I am expecting to be able to see my page in a Browser preview, but am not. Am I missing something? I noticed a message saying "Page is not visible, ordering disabled" in the Advanced tab even after I have set Published to true.

**Comment by rhukster** [comment]
I'm not seeing this behavior.  Can you take a little video of your steps so I can replicate exactly?

**Comment by paulhibbitts** [comment]
Sure thing, I *think* I am doing the expected steps...

https://www.dropbox.com/sh/ny3yl2q90ttzpbv/AACxg1xJA-YGja1Cj1GguX-ja?dl=0

**Comment by rhukster** [comment]
Ok that helps a tonne!

So everything is working 100% correct, but there are a couple of things your not taking into account, and we can certainly discuss if this can be improved:

1) Your first page was created as expected, page was visible.
2) Your second page you had the "visible: auto" option set in the modal popup

Let's stop here and explain how that works.  At the root your folders have 01. 02, etc prefixes to define order, these automatically ensure a page is visible (I'm sure you remember this).  When you created the first page, the  'visible: auto' option checked to see if folder numbers at root were used, finding yes, it created your new page with a folder prefix.. making it visible.  The second page, was a new page and as such with no other sibling pages, it defaulted to no folder prefix (like a blog post would be), so although it is published is not visible in the menu (you dont want blog posts showing up in the menu.

Brings me to a another point. For things to sub-pages to show up in the menu of antimatter, you must enable the dropdown in the theme preferences. 

So in order to have the correct outcome that you are expecting:

1. Enable the dropdown menu in Antimatter options
2. Create the subpage and either:
    a. Set visible option to **Yes**
    b. Leave it on auto, but then in advanced options choose a) **Enable folder prefix** or b) **Visibility: Enabled**

This would produce the outcome you are looking for.  Some things to consider though:

1. I can't just change the default antimatter dropdown behavior to enabled because it could adversely effect people with antimatter based sites that don't expect this change in behavior. 
2. The visibility "auto" setting was added to make things simpler.  If you are adding a page to a folder level that has visible pages, new pages will be visible.  If you are adding other pages, such as blog posts, the default is not visible in the menu.  The theory here is most pages that show up in a menu are added early with intent and purpose.  The more common type of pages added by you or 3rd parties are going to be blog/news/article type pages.  These are typically not intended to be in a menu, and hence the default visibility state is disable.

**Comment by paulhibbitts** [comment]
Thanks for the detailed background info. I've only used Antimatter theme a bit myself in the past, so I've only got really introductory knowledge of it.

Regarding expected outcomes, I can see that changing the dropdown menu settings could be troublesome so as you described these should be left alone. 

The Visibility "auto" setting behavior is pretty tricky though, esp. for less experienced Grav users and even for those experienced in other themes (as I am). Being "Auto" I assumed it would be defaulted so I could see my pages, but that was wrong. 

Hmmm... AntiMatter is positioned as the default Grav theme, and with the Admin Panel I think we need to make things a bit more obvious at first than the above.

I re-did this workflow, but used the Blog template rather than the Default template (view at https://www.dropbox.com/sh/ny3yl2q90ttzpbv/AACxg1xJA-YGja1Cj1GguX-ja?dl=0), and things worked exactly as expected. Might this point us to a way we can make things easier for new users to the Admin Panel? For example, what if "Blog" was the initial default for creating new Root pages? The current default "Default" seems to be pretty tricky for new Grav Admin users, and when  working without the Admin I've already greatly appreciated how easy it was to figure out how to do things with Grav.

**Comment by rhukster** [comment]
The thing is that Grav is so flexible and intentionally does not impose things upon you. There is also no one-way of doing things, and no standard layout.  If it was a blog-only platform this would be a non-issue, but it's not, it can be anything, so there is really no clear workflow that is going to satisfy everyone.  

I really think the best solution is to offer some videos or improved documentation.  I hate to impose any fixed processes or workflow that primarily focus on novice users when really the target audience is not a raw beginner, but a semi-experienced web developer.  Day-to-day users of Grav can include novice users of course, but i dont think its appropriate for a novice user to be doing ground-up development.

Web development is not a simple thing, no matter what Wix or Squarespace would have you believe, just like software development is not for the uninitiated.  There are too many variables, too many technologies and too many intangible factors that only experienced web professionals will know. 

So in summary, I think development tasks should be focused towards developers, and content creation/edit tasks should be focused towards a broader audience.  I think your use case is a more developer oriented one, although it's a grey area :)

**Comment by paulhibbitts** [comment]
I agree wth what you are saying totally, I just wonder if we still can't help out new(er) users to the Admin Panel a bit more with such as essential content creation process.

Going over your initial reply again, it seems that the "visible: auto" setting is really the core of a lot of my trouble. Just putting this by you, but what if either the default for this was changed or even if that setting was removed? Between Published and Visible users have lots of options to hide/show content and both are pretty quickly understandle. They are also in the core, at least I came across both of theme before the Admin Panel. My gut is that the visibility "auto" setting actually makes things harder, as it's behaviour and results are unclear as compared to Visible and Published flags, what are your thoughts?

**Comment by paulhibbitts** [comment]
Another follow-up comment: will the Admin Panel be included in site skeletons? If so, then perhaps there the "default" page creation behavior might be more specific? I.e. one-page site vs. blog site. Just throwing a few ideas around :-)

**Comment by rhukster** [comment]
I don't have any immediate plans to include the plugin with skeletons. A lot of people won't need or want the admin so I don't want to ram it down their throats :)
